### PR TITLE
[Examples] Added context manager to datasets map

### LINKED ISSUE #12363: [examples] add `main_process_first` context manager to datasets map calls
We need to replay this addition that has been modelled in `run_translation.py` in https://github.com/huggingface/transformers/pull/12351 to all other pytorch examples

The actual changes for the model example are:
https://github.com/huggingface/transformers/pull/12351/files#diff-09777f56cee1060a535a72ce99a6c96cdb7f330c8cc3f9dcca442b3f7768237a
(just `run_translation.py`)

Here is a time-saver:
```
find examples/pytorch -type f -exec perl -0777 -pi -e 's|^(\s+)(train_dataset = train_dataset.map\(.*?\))|x($1, $2)|msge; BEGIN {sub x {($p, $t) = @_ ; $t =~ s/^/    /msg; return qq[${p}with training_args.main_process_first(desc="train dataset map pre-processing"):\n$p$t] } }' {} \;

find examples/pytorch -type f -exec perl -0777 -pi -e 's|^(\s+)(eval_dataset = eval_dataset.map\(.*?\))|x($1, $2)|msge; BEGIN {sub x {($p, $t) = @_ ; $t =~ s/^/    /msg; return qq[${p}with training_args.main_process_first(desc="validation dataset map pre-processing"):\n$p$t] } }' {} \;

find examples/pytorch -type f -exec perl -0777 -pi -e 's|^(\s+)(predict_dataset = predict_dataset.map\(.*?\))|x($1, $2)|msge; BEGIN {sub x {($p, $t) = @_ ; $t =~ s/^/    /msg; return qq[${p}with training_args.main_process_first(desc="prediction dataset map pre-processing"):\n$p$t] } }' {} \;

git checkout examples/pytorch/translation/run_translation.py

make fixup
```
I noticed other scripts may have other `datasets.map` calls, which get automatically rewritten by the scripts above, so please review the changes to see if the `desc` needs to be modified. But we want to use the context manager on all of these calls, it's possible that the perl rewrite scripts didn't catch some.

- also this template needs to have this change as well:
`templates/adding_a_new_example_script/\{\{cookiecutter.directory_name\}\}/run_\{\{cookiecutter.example_shortcut\}\}.py`
can do via perl or manually or whatever other way works for you.

And please validate that scripts still work, by either running:
```
RUN_SLOW=1 pytest  examples/pytorch/test_examples.py
```
or running each script manually as explained in its corresponding `README.md` file.

This issue is open to all and should be very simple to complete, the main effort is to validate.

And thank you for your contribution!

### PR DESCRIPTION
**Pull request by bhadreshpsavani**
# What does this PR do?

Fixes #12363

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@stas00 @sgugger

### CODE REVIEW

File: examples/pytorch/language-modeling/run_mlm.py [hunk 1]
```diff
@@ -429,13 +431,14 @@ def group_texts(examples):
...
         # https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.map
 
-        tokenized_datasets = tokenized_datasets.map(
-            group_texts,
-            batched=True,
-            num_proc=data_args.preprocessing_num_workers,
-            load_from_cache_file=not data_args.overwrite_cache,
-            desc=f"Grouping texts in chunks of {max_seq_length}",
-        )
+        with training_args.main_process_first(desc="dataset map pre-processing"):
```
Comment (sgugger) [inline examples/pytorch/language-modeling/run_mlm.py:434]:
```suggestion
        with training_args.main_process_first(desc="grouping text together"):
```

File: examples/pytorch/language-modeling/run_clm.py [hunk 2]
```diff
@@ -404,13 +405,14 @@ def group_texts(examples):
...
     # https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.map
 
-    lm_datasets = tokenized_datasets.map(
-        group_texts,
-        batched=True,
-        num_proc=data_args.preprocessing_num_workers,
-        load_from_cache_file=not data_args.overwrite_cache,
-        desc=f"Grouping texts in chunks of {block_size}",
-    )
+    with training_args.main_process_first(desc="dataset map pre-processing"):
```
Comment (sgugger) [inline examples/pytorch/language-modeling/run_clm.py:408]:
```suggestion
    with training_args.main_process_first(desc="grouping texts together"):
```

File: tests/sagemaker/scripts/pytorch/run_glue_model_parallelism.py [hunk 3]
```diff
@@ -503,7 +503,7 @@ def compute_metrics(p: EvalPrediction):
 
         for test_dataset, task in zip(test_datasets, tasks):
             # Removing the `label` columns because it contains -1 and Trainer won't like that.
-            test_dataset.remove_columns_("label")
+            test_dataset = test_dataset.remove_columns("label")
```
Comment (sgugger) [inline tests/sagemaker/scripts/pytorch/run_glue_model_parallelism.py:506]: Thanks for fixing!

**Review by sgugger — APPROVED** [review]
Thanks a lot for working on this!

**Review by stas00 — APPROVED** [review]
Great work, @bhadreshpsavani. Thank you!

**Pull request merged**